pkg/query-service/interfaces/interface.go

```diff
@@ -81,6 +81,7 @@ type Reader interface {
 	CheckClickHouse(ctx context.Context) error
 
 	GetMetricMetadata(context.Context, valuer.UUID, string, string) (*v3.MetricMetadataResponse, error)
+	GetFirstSeenFromMetricMetadata(ctx context.Context, lookupKeys []model.MetricMetadataLookupKey) (map[model.MetricMetadataLookupKey]int64, error)
```

Comment:
Let's move from here to metadata store

- https://github.com/SigNoz/signoz/blob/main/pkg/telemetrymetadata/metadata.go
- https://github.com/SigNoz/signoz/blob/main/pkg/types/telemetrytypes/store.go#L11